### PR TITLE
Add Bling Subsystem and Command

### LINKED ISSUE #8: Bling Subsystem and Command in the 2024 codebase
Create the BlingSubsystem.java and the BlingCommand,java to be used by other commands in the codebase.

### PR DESCRIPTION
**Pull request by Patchy-pirate**
## Describe your changes

- [ ] Bug fix
- [x] New feature

Explain what you did:
> i added bling command and bling subsystem and changed the SetBlingCommand to BlingCommand

## Issue ticket (add number)

> https://github.com/FRC2706/2024-2706-Robot-Code/issues/...

## Checklist before requesting a review
- [X] I have performed a self-review of my code.
- [X] I have tested thoroughly.
- [X] I have written documentation and explained about my changes to the best of my ability.

### CODE REVIEW

**Comment by weizhang-2706** [comment]
It looks like this branch has conflicts. First pull from main and then fix the conflicts.

**Comment by GabbyGenereux** [comment]
> It looks like this branch has conflicts. First pull from main and then fix the conflicts.

Done

File: src/main/java/frc/robot/Config.java [hunk 1]
```diff
@@ -255,6 +255,9 @@ public static final class AutoConstants {
         new TrapezoidProfile.Constraints(
             kMaxAngularSpeedRadiansPerSecond, kMaxAngularSpeedRadiansPerSecondSquared);
   }
+   public static final class BlingConstants {
+    public static int CANDLE = 15;
```
Comment (weizhang-2706) [inline src/main/java/frc/robot/Config.java:261]: You can put this line inside class CANID.
(Note: so far we always set CANID of 15 for CANDLE for Beetle and Poseidon.)

File: src/main/java/frc/robot/commands/BlingCommand.java [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+  @Override
+  public void initialize() {
+    if (bling != null)
+    {
+      if (m_blingPatternId!= 0) {
+        bling.setBrightness();
+      }
+      switch( m_blingPatternId )
+      {
+        case 0:
```
Comment (weizhang-2706) [inline src/main/java/frc/robot/commands/BlingCommand.java]: Could have an enum for these integers. DISABLE for 0, PURPLE for 1, ..., etc.

**Review by weizhang-2706 — CHANGES_REQUESTED** [review]

File: src/main/java/frc/robot/Config.java (hunk 1 above)
Comment (GabbyGenereux) [reply src/main/java/frc/robot/Config.java:261]: After discussion at the meeting on Wednesday we have decided to keep the structure of having each mechanism have its own class, where the CANIDs (and any other constants) will live.

File: src/main/java/frc/robot/commands/BlingCommand.java (hunk 2 above)
Comment (Patchy-pirate) [reply src/main/java/frc/robot/commands/BlingCommand.java]: I have changed the SetBlingCommand command to an enum under the name BlingColour, as per requested.

**Review by weizhang-2706 — APPROVED** [review]

**Review by ErikCald — APPROVED** [review]
!evorppA

**Pull request merged**